cmd/xl-v1-object.go

```diff
@@ -593,7 +597,13 @@ func (xl xlObjects) PutObject(bucket string, object string, size int64, data io.
 
 		// If we didn't write anything or we know that the next part doesn't have any
 		// data to write, we should quit this loop immediately
-		if partSizeWritten == 0 || getPartSizeFromIdx(size, globalPutPartSize, partIdx+1) == 0 {
+		var partSize int64
+		partSize, err = getPartSizeFromIdx(size, globalPutPartSize, partIdx+1)
+		if err != nil {
+			return ObjectInfo{}, toObjectErr(err, bucket, object)
+		}
+
+		if partSizeWritten == 0 || partSize == 0 {
```

Comment:
Can you make the check `if partSizeWritten == 0`  separately just before calculating the part size of the next part (and update comments) ? it would be easier to read. LGTM otherwise.